okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/MultiServerRetryInterceptor.java

```diff
@@ -46,6 +47,8 @@ public Response intercept(Chain chain) throws IOException {
         // Try original request first.
         try {
             return chain.proceed(chain.request());
+        } catch (InterruptedIOException e) {
```

Comment:
where would this exception get thrown?

Reply:
This gets thrown by OkHttp if you interrupt the thread before OkHttp tries to connect to the socket. Currently, http-remoting will not release the thread, which is sadness. However, the window for this to happen is very small - I've updated the PR to reflect a more serious/likely scenario, which is when the thread is interrupted while waiting for a socket response. See the failing test.

Reply:
(Rethrowing the InterruptedIOExceptions does not solve the "interrupted while waiting for a socket repsonse" problem - diving into the code, it looks like it's blocking on socket operations by OkHttp)